javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -74,10 +73,11 @@ private static int treeSize(int branchCount, int depthShift) {
         return new BitMappedTrie<>(array, 0, size, shift);
     }
 
-    BitMappedTrie<T> prepend(T leading) {
-        final int newSize = length() + 1;
+    BitMappedTrie<T> prepend(Object[] leading) {
```

Comment:
now we only modify the tree (via append and prepend), if we have full arrays, not one-by-one